test_crates/trait_method_added/old/src/lib.rs

```diff
@@ -0,0 +1,30 @@
+mod sealed {
+    pub(crate) trait Sealed {}
+}
+
+// ---- Should be reported ----
+pub trait WillGainMethodWithoutDefault {}
+
+pub trait WillGainAnotherMethodWithoutDefault {
+    fn one_method(self);
+}
+
+pub trait WillGainMultipleMethodsWithoutDefault {}
+
+// ---- Should not be reported ----
+pub trait WillGainMethodWithDefault {}
+
+pub trait WillGainAnotherMethodWithDefault {
+    fn one_method(self);
+}
+
+pub trait WillGainMethodWithoutDefaultSealed: sealed::Sealed {}
+
+pub trait WillGainMethodWithoutDefaultAndLoseSeal: sealed::Sealed {}
+
+/*
+Will let this case to be reported only by the newly sealed trait Lint,
+and not by this one, since sealing a trait indicates that the user 
+wants to remove this type from the public API.
+*/
```

Comment:
This is correct, and like I said earlier, it's a judgment call that could go either way.

The opposite argument is that the user might not have realized that their trait is sealed (it has become so by accident). In that case, since it wasn't an intentional decision, items requiring implementation are a separate and unrelated issue.